apisx/v1alpha2/backendtrafficpolicy.go

```diff
@@ -0,0 +1,150 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+// +kubebuilder:resource:categories=gateway-api,shortName=btrafficpolicy
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+//
+// BackendTrafficPolicy is a Direct Attached Policy.
+// +kubebuilder:metadata:labels="gateway.networking.k8s.io/policy=Direct"
+
+// BackendTrafficPolicy defines the configuration for how traffic to a
+// target backend should be handled.
+type BackendTrafficPolicy struct {
+	// Support: Extended
+	//
+	// +optional
+	// <gateway:experimental>
+
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of BackendTrafficPolicy.
+	Spec BackendTrafficPolicySpec `json:"spec"`
+
+	// Status defines the current state of BackendTrafficPolicy.
+	Status PolicyStatus `json:"status,omitempty"`
+}
+
+// BackendTrafficPolicyList contains a list of BackendTrafficPolicies
+// +kubebuilder:object:root=true
+type BackendTrafficPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []BackendTrafficPolicy `json:"items"`
+}
+
+// BackendTrafficPolicySpec define the desired state of BackendTrafficPolicy
+// Note: there is no Override or Default policy configuration.
+type BackendTrafficPolicySpec struct {
+	// TargetRef identifies an API object to apply policy to.
+	// Currently, Backends (i.e. Service, ServiceImport, or any
+	// implementation-specific backendRef) are the only valid API
+	// target references.
+	//
+	// +listType=map
+	// +listMapKey=group
+	// +listMapKey=kind
+	// +listMapKey=name
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=16
+	TargetRefs []LocalPolicyTargetReference `json:"targetRefs"`
+
+	// RetryConstraint defines the configuration for when to allow or prevent
+	// further retries to a target backend by dynamically calculating a 'retry
+	// budget'. This budget is calculated based on the percentage of incoming
+	// traffic composed of retries over a given time interval. Once the budget
+	// is exceeded, additional retries will be rejected by the backend.
+	//
+	// For example, if the retry budget interval is 10 seconds, there have been
+	// 1000 active requests in the past 10 seconds, and the allowed percentage
+	// of requests that can be retried is 20% (the default), then 200 of those
+	// requests may be composed of retries. Active requests will only be
+	// considered for the duration of the interval when calculating the retry
+	// budget.
+	//
+	// Configuring a RetryConstraint in BackendTrafficPolicy is compatible with
+	// HTTPRoute Retry settings for each HTTPRouteRule that targets the same
```

Comment:
By applying this at HTTPRoute level, Envoy-based implementations are not going to have a natural mapping, since this is implemented at Cluster level in Envoy, not route level. Envoy would need to implement this at route level as well to make this feasible.